helix-core/src/main/java/org/apache/helix/manager/zk/GenericSerializer.java

```diff
@@ -0,0 +1,29 @@
+package org.apache.helix.manager.zk;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * The generic interface serialize/deserialize custom data type of T to/from byte[]
+ * @param <T> The custom data type
+ */
+public interface GenericSerializer<T> {
+  byte[] serialize(T data);
+  T deserilize(byte[] content);
+}
```

Comment:
This serializer interface is not needed. Let us remove it.

It will have to be either PathBasedZkSerializer (Helix library) or ZkSerializer (ZK library). Let me confirm and update here in this PR.

Reply:
Sure, thanks for the confirmation!